tensorrt_llm/_torch/disaggregation/native/transfer.py

```
@@ -2095,6 +2659,74 @@ def receive(self, slice: KVSlice) -> None:
+    def cancel_unpublished_task(self, task: KVRecvTask) -> bool:
+        cancelled = task.cancel_unpublished()
+        if self._aux_physical_owner is not None:
+            self._aux_physical_owner.cancel_unpublished()
+        return cancelled
+
+    def _accept_candidate_writer(self, peer_rank: int, succeeded: bool) -> bool:
+        if not self._candidate_by_writer:
+            return True
+        cohort = self._candidate_by_writer.get(peer_rank)
+        selected = self._selected_writer_cohort
+        ambiguous = cohort is None or selected is None and not succeeded
```

Ruff (0.16.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)